src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -176,6 +182,38 @@ def create(cls, catalog: CatalogName) -> 'RepositoryPlugin':
     def sources(self) -> AbstractSet[TDRSourceSpec]:
         return self._sources
 
+    @cached_property
```

Comment:
State shouldn't be mutable unless it has to be. This is even more so the case for a cache, which is meant to be shared and therefore would be very exotic if it were mutable.